saleor/graphql/product/mutations/products.py

```diff
@@ -1198,106 +1202,122 @@ def perform_mutation(cls, _root, info, **data):
             info, data["product"], field="product", only_type=Product
         )
 
-        image_data = info.context.FILES.get(data["image"])
-        validate_image_file(image_data, "image")
+        image = data.get("image")
+        video_url = data.get("video_url")
+        if not image and not video_url:
+            raise ValidationError("Image or video is required.")
+        if image and video_url:
+            raise ValidationError("Only one is required.")
+
+        if image and not video_url:
+            image_data = info.context.FILES.get(image)
+            validate_image_file(image_data, "image")
+            media = product.media.create(
+                image=image_data, alt=data.get("alt", ""), type=ProductMediaTypes.IMAGE
+            )
+            create_product_thumbnails.delay(media.pk)
+
+        if video_url and not image:
```

Comment:
There is an option that we will have video and image? I think you validated this case  on 1209 line.